forward_command_controller/include/forward_command_controller/forward_joint_group_command_controller.h

```diff
@@ -106,33 +99,87 @@ class ForwardJointGroupCommandController: public controller_interface::Controlle
 
     commands_buffer_.writeFromNonRT(std::vector<double>(n_joints_, 0.0));
 
-    sub_command_ = n.subscribe<std_msgs::Float64MultiArray>("command", 1, &ForwardJointGroupCommandController::commandCB, this);
     return true;
   }
 
-  void starting(const ros::Time& time);
-  void update(const ros::Time& /*time*/, const ros::Duration& /*period*/)
+  virtual void starting(const ros::Time& time);
+  virtual void update(const ros::Time& /*time*/, const ros::Duration& /*period*/)
   {
     std::vector<double> & commands = *commands_buffer_.readFromRT();
     for(unsigned int i=0; i<n_joints_; i++)
     {  joints_[i].setCommand(commands[i]);  }
   }
 
+protected:
   std::vector< std::string > joint_names_;
   std::vector< hardware_interface::JointHandle > joints_;
   realtime_tools::RealtimeBuffer<std::vector<double> > commands_buffer_;
   unsigned int n_joints_;
+};
+
+template <> inline void ForwardJointGroupCommandControllerBase<hardware_interface::EffortJointInterface>::starting(const ros::Time& time)
```

Comment:
Moving them into `forward_command_controller` allows to reuse them as well.
Most users will be fine with the default for `starting()`, if not they can reimplement it, it is (and has been) virtual.

Your examples
```
effort_controllers::JointGroupPositionController
effort_controllers::JointGroupVelocityController
```
are no forward controllers and should not use this base class.